valhalla/proposals/tasks.py

```diff
@@ -1,29 +1,39 @@
 from __future__ import absolute_import, unicode_literals
+from valhalla.userrequests.models import UserRequest, Request
+
+from django.conf import settings
 from celery import shared_task
+import requests
 import logging
-from valhalla.proposals.models import Semester, TimeAllocation
-from valhalla.proposals.accounting import get_time_totals_from_pond
 
 logger = logging.getLogger(__name__)
 
 
-@shared_task
-def update_time_allocation(time_allocation_id):
-    talloc = TimeAllocation.objects.get(pk=time_allocation_id)
-    logger.info('Updating timeallocation for %s', talloc.proposal, extra={'tags': {'proposal': talloc.proposal.id}})
-    std_total = get_time_totals_from_pond(talloc, talloc.semester.start, talloc.semester.end, too=False)
-    too_total = get_time_totals_from_pond(talloc, talloc.semester.start, talloc.semester.end, too=True)
-    talloc.std_time_used = std_total
-    talloc.too_time_used = too_total
-    talloc.save()
-
+def query_pond_for_time_used(request):
+    logger.info('Attempting to get time used for %s', request.id)
+    url = '{0}/pond/pond/accounting/{1}/'.format(settings.POND_URL, request.id)
+    response = requests.get(url)
+    response.raise_for_status()
+    return {
+        'too_time': response.json()['block_bounded_attempted_hours'],
+        'std_time': response.json()['attempted_hours']
+    }
 
-@shared_task
-def run_accounting(semesters=None):
-    if not semesters:
-        semesters = Semester.current_semesters()
 
-    for semester in semesters:
-        logger.info('Performing accounting for semester: %s', semester)
-        for talloc in semester.timeallocation_set.all():
-            update_time_allocation.delay(talloc.id)
+@shared_task()
+def run_accounting():
+    for request in Request.need_accounting():
+        log_tags = {'tags': {'proposal': request.user_request.proposal.id}}
+        try:
+            talloc = request.timeallocation
+            logger.info('Performing accounting for request: %s', request.id, extra=log_tags)
+            time_used = query_pond_for_time_used(request)
+            if request.user_request.observation_type == UserRequest.TOO:
+                talloc.too_time_used += time_used['too_time']
+            else:
+                talloc.std_time_used += time_used['std_time']
+            talloc.save()
+            request.accounted = True
+            request.save()
```

Comment:
I think this should be wrapped in @transaction.atomic like the functions that modify request state in state_changes.py are.

Reply:
Good idea.